radix-engine/src/engine/kernel.rs

```diff
@@ -632,7 +616,7 @@ where
                 node_pointer
                     .acquire_lock(substate_id.clone(), false, false, &mut self.track)
                     .map_err(RuntimeError::KernelError)?;
-                let node_ref = node_pointer.to_ref(&self.call_frames, &mut self.track);
+                let mut node_ref = node_pointer.to_ref(&self.call_frames, &mut self.track);
```

Comment:
As discussed, this mainly because we need to update the cache of the track. If we move the node initialisation to when a reference is created, we will be able to get rid of the `mut` requirement.